Sources/ConnectionPoolModule/PoolStateMachine+ConnectionState.swift

```diff
@@ -409,7 +409,11 @@ extension PoolStateMachine {
                     self.state = .leased(connection, usedStreams: newUsedStreams, maxStreams: maxStreams, keepAlive: keepAlive)
                     return .leased(availableStreams: availableStreams)
                 }
-            case .backingOff, .starting, .idle, .closing, .closed:
+
+            case .closing:
+                return nil
```

Comment:
Now that we keep a record of all the connections during the shutdown It is now possible to catch a connection in the closing state when release is called after triggerForceShutdown